pkg/aws/helpers.go

```diff
@@ -535,18 +537,15 @@ func GetOperatorRolePolicyPrefixFromCluster(cluster *cmv1.Cluster, awsClient Cli
 		return rankedPolicyPrefix[0], nil
 	}
 
-	// If no commonly used prefix is found use operator role prefix
-	operatorRoleName, err := GetResourceIdFromARN(operatorRoles[0].RoleARN())
-	if err != nil {
-		return "", err
-	}
-	index := strings.LastIndex(operatorRoleName, "-openshift")
-	if index != -1 {
-		return operatorRoleName[0:index], nil
+	//If no standard prefix is found, tries to look for the longest common prefix
+	// TODO: check if it makes sense to only use this and remove "-openshift" later
+	policyPrefix := helper.LongestCommonPrefixBySorting(policyNames)
+	if policyPrefix != "" {
+		return policyPrefix, nil
 	}
 
-	// If nothing works setup a new prefix from name and random label so as to not interrupt user flow
-	return fmt.Sprintf("%s-%s", cluster.Name(), helper.RandomLabel(4)), nil
+	// If nothing works uses operator role prefix as to not interrupt flow
+	return cluster.AWS().STS().OperatorRolePrefix(), nil
```

Comment:
Maybe that should get precedence over LCP?
Only if this value is empty somehow we can fallback to LCP

Reply:
The operator policy prefix is usually based on the account roles prefix rather than the operator role prefix.
That's why I think LCP should take precedence